ios/engine/KMEI/KeymanEngine/Classes/KeymanPackage.swift

```diff
@@ -24,11 +24,32 @@ public enum KMPError : String, Error {
 public class KeymanPackage {
   static private let kmpFile = "kmp.json"
   public let sourceFolder: URL
+  public let id: String
   internal let metadata: KMPMetadata
+  internal let isTemp: Bool
 
   internal init(metadata: KMPMetadata, folder: URL) {
     sourceFolder = folder
     self.metadata = metadata
+
+    let folderName = folder.lastPathComponent
+    let cacheDirectory = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)[0]
+    var nameComponents = folderName.components(separatedBy: ".")
+
+    // Have we parsed a temporary extraction site?
+    // This case arises during package installation.
+    isTemp = folderName.hasSuffix(".kmp.zip") && folder.path.contains(cacheDirectory.path)
```

Comment:
It seems that this won't remove `.kmp`, only `.kmp.zip`?

Reply:
Correct.  The zipping utility we use requires a `.zip` extension, so we rename `.kmp`s to `.kmp.zip` to satisfy it.  Correspondingly, the extracted folder will end in `.kmp.zip`.

My current plan is to preserve any downloaded `.kmp`s anyway, storing them in the app's Documents folder for ad-hoc sharing if desired.  _Could_ make that an option on the install window once we're ready for that upcoming rework, and it's still just a plan.